### PR TITLE
[Sync Pipeline Inference] Sync pipeline inference branch to main 

### PR DESCRIPTION
**Pull request by FoolPlayer**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [ ] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/pipeline/schedule/generate.py [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+        Receive the output from previous stage
+
+        Returns:
+            Any: The output from previous stage
+        '''
+        if self.stage_manager.num_stages == 2:
+            return self.comm.p2p_recv()
+        return self.comm.recv_forward()
+
+    def LoadStageAction(self, model: Module) -> None:
```
Comment (ver217) [inline colossalai/pipeline/schedule/generate.py:130]: The naming is not standard. Method name should not be camel case.

File: tests/test_generate/test_pipeline_infer.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
+from copy import deepcopy
```
Comment (ver217) [inline tests/test_generate/test_pipeline_infer.py:1]: Move this to `test_inference/`

**Review by ver217 — APPROVED** [review]

File: colossalai/pipeline/schedule/generate.py (hunk 1 above)
Comment (FoolPlayer) [reply colossalai/pipeline/schedule/generate.py]: ✅

File: tests/test_generate/test_pipeline_infer.py (hunk 2 above)
Comment (FoolPlayer) [reply tests/test_generate/test_pipeline_infer.py]: done

**Review by CjhHa1 — APPROVED** [review]

**Pull request merged**